File: lib/model/body.dart
import 'body_type.dart';
import 'gallery.dart';

/// A celestial object with a bunch of characteristics
class Body {
  final String id;
  final String name;
  final String englishName;
  final bool isPlanet;
  final List<String>? moons;
  final int semimajorAxis;
  final int perihelion;
  final double eccentricity;
  final double inclination;
  final double? mass;
  final double? vol;
  final double density;
  final double gravity;
  final double escape;
  final double meanRadius;
  final double equaRadius;
  final double polarRadius;
  final double flattening;
  final String dimension;
  final double sideralOrbit;
  final double sideralRotation;
  final String? aroundPlanet;
  final String discoveredBy;
  final String discoveryDate;
  final String alternativeName;
  final double axialTilt;
  final int avgTemp;
  final double mainAnomaly;
  final double argPeriapsis;
  final double longAscNode;
  final BodyType? bodyType;
  final Gallery gallery;

  Body({
    required this.id,
    required this.name,
    required this.englishName,
    required this.isPlanet,
    required this.moons,
    required this.semimajorAxis,
    required this.perihelion,
    required this.eccentricity,
    required this.inclination,
    required this.mass,
    required this.vol,
    required this.density,
    required this.gravity,
    required this.escape,
    required this.meanRadius,
    required this.equaRadius,
    required this.polarRadius,
    required this.flattening,
    required this.dimension,
    required this.sideralOrbit,
    required this.sideralRotation,
    required this.aroundPlanet,
    required this.discoveredBy,
    required this.discoveryDate,
    required this.alternativeName,
    required this.axialTilt,
    required this.avgTemp,
    required this.mainAnomaly,
    required this.argPeriapsis,
    required this.longAscNode,
    required this.bodyType,
    required this.gallery,
  });

  @override
  String toString() {
    return "Nom : $name\n"
        "Type d'astre : $bodyType\n"
        "Nombre d'images : $gallery\n\n";
  }
}

File: lib/python/body.py
from dataclasses import dataclass
from typing import Optional, List

from models.body_type import BodyType
from models.gallery import Gallery


@dataclass
class Body:
    _id: str
    _name: str
    _english_name: str
    _is_planet: bool
    _moons: Optional[List[str]]
    _semimajor_axis: int
    _perihelion: int
    _aphelion: int
    _eccentricity: float
    _inclination: float
    _mass: float
    _vol: float
    _density: float
    _gravity: float
    _escape: float
    _mean_radius: float
    _equa_radius: float
    _polar_radius: float
    _flattening: float
    _dimension: str
    _sideral_orbit: float
    _sideral_rotation: float
    _around_planet: str
    _discovered_by: str
    _discovery_date: str
    _alternative_name: str
    _axial_tilt: float
    _avg_temp: int
    _main_anomaly: float
    _arg_periapsis: float
    _long_asc_node: float
    _body_type: BodyType
    # Pictures getted from NASA's API
    _gallery: Optional[Gallery]
